### ISSUE TITLE
Plan fails if using SPN and the management group has no subscriptions

### ISSUE THREAD

**Issue by teemukom** [Azure/enterprise-azure-policy-as-code#442]
SPN created which has Owner permissions on epac-dev root and the necessary Graph permissions

Plan fails when processing policy assignments
```
===================================================================================================
Processing Policy Assignments JSON files in folder './/Definitions/policyAssignments'
===================================================================================================
Loaded Module 'Az.Accounts'
Loaded Module 'Az.MSGraph'
Loaded Module 'Az.Accounts'
Loaded Module 'Az.Authorization'
Get-AzRoleDefinition: /home/vsts/work/1/s/Scripts/Helpers/Build-AssignmentPlan.ps1:20
Line |
  20 |      $roleDefinitionList = Get-AzRoleDefinition
     |                            ~~~~~~~~~~~~~~~~~~~~
     | Object reference not set to an instance of an object.

##[error]PowerShell exited with code '1'.

```
Testing it interactive the error message is more clear although the script continues to process:
```
===================================================================================================
Processing Policy Assignments JSON files in folder '..\..\Definitions\/policyAssignments'
===================================================================================================
Get-AzRoleDefinition: \Scripts\Helpers\Build-AssignmentPlan.ps1:20
Line |
  20 |      $roleDefinitionList = Get-AzRoleDefinition
     |                            ~~~~~~~~~~~~~~~~~~~~
     | No subscription was found in the default profile and no scope was specified. Either specify a scope or use a tenant with a subscription to run the command.
```

Similar issue already in the Azure powershell repo https://github.com/Azure/azure-powershell/issues/22362

**Comment by techlake** [comment]
Since we are planning a new approach (see #438 ), we will prioritize that enhancement.

**Comment by techlake** [comment]
I should have this by next week.

**Comment by techlake** [comment]
Fixed in v8.5.0
